src/wazuh_modules/agent_upgrade/manager/wm_agent_upgrade_validate.c

```diff
@@ -415,3 +415,18 @@ bool wm_agent_upgrade_validate_task_status_message(const cJSON *response, char *
     }
     return false;
 }
+
+bool wm_agent_upgrade_validate_task_ids_message(const cJSON *input_json, int *agent_id) {
+    cJSON *agent_json = cJSON_GetObjectItem(input_json, "agent");
+    cJSON *data_json = cJSON_GetObjectItem(input_json, "data");
+    cJSON *task_json = cJSON_GetObjectItem(input_json, "task_id");
+
+    if (agent_json && (agent_json->type == cJSON_Number) && data_json && (data_json->type == cJSON_String)) {
+        agent_id = agent_json->valueint;
+
+        if (task_json && (task_json->type == cJSON_Number)) {
```

Comment:
`task_id` may not be in the JSON but we should also return `true` in that case so we remove just this agent from the hash table.